internal/db/schema/migrations/postgres/16/01_plugin.up.sql

```diff
@@ -0,0 +1,149 @@
+begin;
+
+  -- We are updating the plugin table here to have a scope id, since
+  -- subtypes of plugin should be scoped.  We use 'global' as the default
+  -- just for the purpose of the migration since the scope cannot be null
+  -- and the one plugin that already exists (pi_system) can be globally
+  -- scoped.  The immediately following statement removes this default.
+  -- We add a name so we can enforce that all plugins have a unique name
+  -- in a specific scope across all plugin subtypes.
+  alter table plugin
+    add column name wt_name,
+    add column scope_id wt_scope_id
+      not null
+      default 'global'
+      references iam_scope(public_id)
+        on delete cascade
+        on update cascade;
+
+  alter table plugin
+    alter column scope_id drop default;
+
+  -- Add constraints that enforce names are unique across all plugin subtypes
+  -- in a specific scope.
+  alter table plugin
+    add constraint plugin_scope_id_public_id_uq
+      unique (scope_id, public_id),
+    add constraint plugin_scope_id_name_uq
+      unique (scope_id, name);
+
+
+  -- insert, update, and delete plugin_subtypes are created since we are adding
+  -- subtyped plugins and we need to keep the base table plugin in sync with all
+  -- subtype tables.
+  create or replace function insert_plugin_subtype()
```

Comment:
In general, we should not use `create or replace` statements because they can hide errors. If we are replacing something, we should know what we are replacing because there are probably dependencies on it and our code needs to account for that. So we should have a `drop` statement followed by a `create` statement.  If either returns an error, there is a problem that needs to be fixed. 

Likewise, if we are creating something new, we should only have a `create` statement. If something already exists, it means we didn't know it existed and the migration should error out.